### PR TITLE
[Draft] pytest migration

### PR DESCRIPTION
**Pull request by canwaf**
* installation of pytest, black, flake8 (Pipenv associated changes)
* rename of testing files
* path fun in test_baseunit.py
* test_infojsontests.py now is classless and migrated to pytest
* all other test files imports updated, classes yote
* #93, #94

### CODE REVIEW

File: csvqb/Pipfile [hunk 1]
```diff
@@ -5,6 +5,7 @@ verify_ssl = true
 
 [dev-packages]
 csvwlib-devtools = {editable = true,path = "./../devtools"}
+ipykernel = "*"
```
Comment (robons) [inline csvqb/Pipfile:8]: What's this for?

File: csvqb/csvqb/.vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
+    "configurations": [
+        {
```
Comment (robons) [inline csvqb/csvqb/.vscode/launch.json:7]: I generally think it's a good idea to keep the user's IDE configuration files out of source control, else developers are always fighting over what their settings are and you end up with loads of conflicts. Perhaps we should add `.vscode` to `.gitignore`?

File: csvqb/csvqb/tests/unit/test_baseunit.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+from typing import List
+import json
+from pathlib import Path
+
+
+def get_test_base_dir() -> Path:
+    path_parts = Path(".").absolute().parts
+    test_index = path_parts.index("tests")
+    test_root_path = Path(*path_parts[0 : test_index + 1])
+    return test_root_path
```
Comment (robons) [inline csvqb/csvqb/tests/unit/test_baseunit.py:12]: Didn't you talk about wanting to do some kind of improvement here? I wrote a mad function to find the `test-cases` directory somewhere in one of the behave files.

Comment (robons) [reply csvqb/csvqb/tests/unit/test_baseunit.py:12]: https://github.com/GSS-Cogs/csvwlib/pull/103/files#diff-5e98b084209290860da165f7a1746943e7ee019ae42045cdeded1126f55f86fdR33

**Review by robons — APPROVED** [review]
Get rid of the `.vscode` files and I'm happy enough for this to go through.

**Comment by canwaf** [comment]
Huzzah. Monster refactor closes #93, #94

**Pull request merged**